src/signature.rs

```diff
@@ -1,7 +1,9 @@
 use fuel_types::Bytes64;
 
-use core::fmt;
 use core::ops::Deref;
+use core::{fmt, str::FromStr};
+
+use crate::Error;
```

Comment:
Just a really small nit, but in the crate we follow these style directives:

- crate imports goes first, third-party libs (including our deps) goes second, and stdlib goes third
- traits are imported only if used

This way, we should

```rust
use crate::Error;

use fuel_types::Bytes64;

use core::ops::Deref;
use core::{fmt, str};
```

and

```rust
impl str::FromStr for Signature {
```